relay-spans/src/span.rs

```diff
@@ -0,0 +1,447 @@
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use serde::Deserialize;
+use serde_repr::Deserialize_repr;
+
+use relay_event_schema::protocol::{Span as EventSpan, SpanId, SpanStatus, Timestamp, TraceId};
+use relay_protocol::{Annotated, Object, Value};
+
+use crate::otel_to_sentry_tags::OTEL_TO_SENTRY_TAGS;
+use crate::status_codes;
+
+/// This is a serde implementation of <https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto>.
+/// A Span represents a single operation performed by a single component of the system.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Span {
```

Comment:
Should we call this type `OtelSpan`? Just to be super clear what it is. Then at a later point, when we invert the pipeline and convert event spans into otel spans, we can rename this type back to `Span`.